src/main/java/codes/biscuit/skyblockaddons/listeners/PlayerListener.java

```diff
@@ -326,6 +326,30 @@ public void onTick(TickEvent.ClientTickEvent e) {
         }
     }
 
+    private ArrayList<ItemStack> previousInventory = null;
+
+    @SubscribeEvent
+    public void onPlayerTick(TickEvent.PlayerTickEvent e) {
+
+        if (main.getUtils().isOnSkyblock() && main.getConfigValues().isEnabled(Feature.BAIT_LIST)) {
+            if (previousInventory == null)
+                previousInventory = new ArrayList<ItemStack>(Arrays.asList(e.player.inventory.mainInventory));
+
+            if (BaitListManager.getInstance().holdingRod()) {
+
+                ArrayList<ItemStack> currentInventory = new ArrayList<ItemStack>(Arrays.asList(e.player.inventory.mainInventory));
+
+                if (!currentInventory.equals(previousInventory)) {
+                    BaitListManager.getInstance().refreshBaits();
+                }
+
+                previousInventory = currentInventory;
+            }
+
+            BaitListManager.getInstance().compareHeldItems(e.player.getHeldItem());
+        }
```

Comment:
This looks like it belongs in `utils.InventoryUtils`. That way you can utilize the Item Pickup Log for watching bait changes.

Reply:
I thought about doing that, but then if Feature.ITEM_PICKUP_LOG is disabled it looks like getInventoryDifference is never used. It also looks like it's specifically made for the log itself. Can you elaborate please?

Reply:
You are right, right now it only checks for differences when the pickup log is enabled, as seen in [PlayerListener.java](https://github.com/BiscuitDevelopment/SkyblockAddons/blob/development/src/main/java/codes/biscuit/skyblockaddons/listeners/PlayerListener.java#L344). You could add the check for your feature there.

Indeed, it seems like in the long run the current logic of the pickup log should be split up into a difference checker and the Feature.ITEM_PICKUP_LOG for your feature to integrate in a simple way.

Reply:
I modified the ItemPickupLog a bit, and properly implemented the PlayerTickEvent (Realised that it actually fires for other players too, but that should have been obvious haha). Could you check again please?